src/Microsoft.Crank.Controller/JobConnection.cs

```diff
@@ -1013,11 +1013,32 @@ private static void CleanTemporaryFiles()
             }
         }
 
-        public async Task<string> DownloadBuildLog()
+        public async Task DownloadBuildLogAsync()
         {
-            var uri = Combine(_serverJobUri, "/buildlog");
+            if (!Job.Options.DownloadBuildLog)
+            {
+                return;
+            }
+
+            var fileName = Path.Combine(Job.Options.DownloadFilesOutput ?? ".", $"{Job.RunId}.{_jobName}.{_serverUri.Host}.{Job.Id}.build.log");
```

Comment:
I think this should handle filenames passed directly. And otherwise use the job name and datetime. This logic is already used in other locations that download files, like traces.